src/scripts/services/okta-service.ts

```diff
@@ -3,13 +3,14 @@ import * as TE from 'fp-ts/lib/TaskEither';
 import { pipe } from 'fp-ts/lib/function';
 import fetch from 'node-fetch';
 import { oktaAPIErrorSchema, urlSchema } from '../../schema';
+import { ReadonlyURL } from 'readonly-types';
 
 /**
  * Parses a url to check whether it is valid
  * @param url - the url to parse
  * @returns a TaskEither that resolves to the url if it is valid, otherwise an error.
  */
-export const parseUrl = (url: string): TE.TaskEither<Error, string> => {
+export const parseUrl = (url: string): TE.TaskEither<Error, ReadonlyURL> => {
```

Comment:
**Question:** since this now returns a `ReadonlyURL` do you have to change the types of the org url parameter in `index.ts` and anywhere else?

Reply:
![image](https://github.com/agiledigital-labs/oktagon/assets/124639768/cd9ee6c7-fb33-4a4c-8081-773af4abbac1)
Its only being used here to either throw or return true so no i don't think so